### PR TITLE
Query suggestions for `GroupByMetrics`

### PR DESCRIPTION
**Pull request by courtneyholcomb**
### Description
Enables suggestions for `GroupByMetrics`. This just means that, if you query an invalid group by metric, the suggestion generator will show you valid group by metrics to help you figure out the right filter.
More detail is provided in commit messages / commit descriptions.

### CODE REVIEW

File: metricflow-semantics/tests_metricflow_semantics/query/test_query_parser.py [hunk 1]
```diff
@@ -640,3 +640,11 @@ def query_parser_from_yaml(yaml_contents: List[YamlConfigFile]) -> MetricFlowQue
     return MetricFlowQueryParser(
         semantic_manifest_lookup=semantic_manifest_lookup,
     )
+
+
+def test_invalid_group_by_metric(bookings_query_parser: MetricFlowQueryParser) -> None:
+    """Tests that a query for an invalid group by metric gives an appropriate group by metric suggestion."""
+    with pytest.raises(InvalidQueryException, match="Metric\\('bookings', group_by=\\['listing'\\]\\)"):
+        bookings_query_parser.parse_and_validate_query(
+            metric_names=("bookings",), where_constraint_str="{{ Metric('listings', ['garbage']) }} > 1"
```
Comment (tlento) [inline metricflow-semantics/tests_metricflow_semantics/query/test_query_parser.py:649]: LOL. Where's the Oscar the Grouch garbage can emoji when I need one...

File: metricflow-semantics/tests_metricflow_semantics/snapshots/test_suggestions.py/str/test_suggestions_for_metric__result_0.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ Error #1:
     The given input does not exactly match any known metrics.
 
     Suggestions:
-      ['bookings', 'booking_fees', 'booking_value', 'instant_bookings', 'booking_payments', 'max_booking_value']
+      ['bookings', 'booking_fees', 'booking_value', 'booking_payments', 'instant_bookings', 'booking_value_p99']
```
Comment (tlento) [inline metricflow-semantics/tests_metricflow_semantics/snapshots/test_suggestions.py/str/test_suggestions_for_metric__result_0.txt:9]: So max_booking_value was tied with booking_value_p99 here and the sort pushed the latter into 6th place?

**Review by tlento — APPROVED** [review]
Seems reasonable!

File: metricflow-semantics/tests_metricflow_semantics/snapshots/test_suggestions.py/str/test_suggestions_for_metric__result_0.txt (hunk 2 above)
Comment (courtneyholcomb) [reply metricflow-semantics/tests_metricflow_semantics/snapshots/test_suggestions.py/str/test_suggestions_for_metric__result_0.txt:9]: Correct!

Comment (courtneyholcomb) [reply metricflow-semantics/tests_metricflow_semantics/snapshots/test_suggestions.py/str/test_suggestions_for_metric__result_0.txt:9]: Well it's actually like a 7-way tie between a bunch of metrics.

**Pull request merged**